languages/thingtalk/load-thingpedia.ts

```diff
@@ -995,6 +995,55 @@ export class ThingpediaLoader {
             span,
             {}
         ));
+
+        for (const arg of q.iterateArguments()) {
+            let op = '==';
+            let vtype : Type[] = [arg.type];
+            const slotOperator = arg.getImplementationAnnotation<string>('slot_operator');
+            if (slotOperator) {
+                op = slotOperator;
+            } else {
+                if (arg.type instanceof Type.Array) {
+                    vtype = [arg.type.elem as Type];
+                    op = 'contains';
+                } else if (arg.type.isRecurrentTimeSpecification) {
+                    vtype = [Type.Date, Type.Time];
+                    op = 'contains';
+                }
+            }
```

Comment:
You need to adjust `op` to be `=~` for String because you're not going through `makeFilter`